Begin Work for state tracking

sithlord48 opened the pull request:
Initial work to track the state of the printer. A few imediate beinfits have already been reaslized you can connect and disconnect now successfully multipal times durring runtime. And simple Pause feature is working.

sithlord48 committed:
Added PrinterState Enum for printer states.
Begin to track the printer state and check it within AtCore.

Signed-off-by: Chris Rizzitello <sithlord48@gmail.com>

patrickelectric commented on testclient/mainwindow.cpp:
```diff
@@ -199,12 +200,37 @@ void MainWindow::fanSpeedPBClicked()
+        break;
+
+    case IDLE:
+        fileName = QFileDialog::getOpenFileName(this, tr("Select A file to print"), QDir::homePath(), "*.gcode");
```
a file

sithlord48 replied:
:heavy_check_mark:  done

patrickelectric commented on testclient/mainwindow.cpp:
```diff
@@ -199,12 +200,37 @@ void MainWindow::fanSpeedPBClicked()
+
+    case BUSY:
+        core->setState(PAUSE);
+        ui->printPB->setText(tr("Resume Print"));
```
Printer

sithlord48 replied:
Its Print as in the current print job running.

patrickelectric commented on testclient/mainwindow.cpp:
```diff
@@ -199,12 +200,37 @@ void MainWindow::fanSpeedPBClicked()
+
+    case PAUSE:
+        core->setState(BUSY);
+        ui->printPB->setText(tr("Pause Print"));
```
Printer

patrickelectric commented on testclient/mainwindow.ui:
```diff
@@ -6,8 +6,8 @@
    <rect>
```
ui changes  in different commit please

sithlord48 replied:
Just the changes to the ui file in a seperate commit?

sithlord48 replied:
:heavy_check_mark:  done

sithlord48 committed:
Prep for use of state based commands.
Removed Disconnect button and clean up the connect area layout.

Signed-off-by: Chris Rizzitello <sithlord48@gmail.com>

sithlord48 committed:
Use Atcore::state() to track printers state.
Gui Button for connect and disconnected combined and functions based on printers state.
Print Button will now change to pause and will pause a print,
and allow for resume after print is paused.

Signed-off-by: Chris Rizzitello <sithlord48@gmail.com>

patrickelectric reviewed (approved):
Can be merged
